rust/BUILD.md

```diff
@@ -0,0 +1,134 @@
+# `evmrs` Development Instructions
+
+## Requirements
+
+- install Rust toolchain >= 1.81.0 from [here](https://rustup.rs/)
+
+## Building evmrs (shared c library)
+
+- build `evmrs` in release mode
+    ```sh
+    cargo build --release # OR run `make` or `make tosca-rust` in Tosca directory
+    ```
+- build `evmrs` with debug symbols
+    ```sh
+    cargo build --profile profiling
+    ```
+
+## Lint
+
+```sh
+cargo clippy --workspace --tests --benches --examples
+```
+
+## Doc
+
+```sh
+cargo doc --workspace --document-private-items --open
+```
+
+## Optimizations
+
+By default, `evmrs` is build with the simplest and most idiomatic version.
+All optimizations are behind feature flags.
+A list of all features can be found in the `[features]` section in [Cargo.toml](./Cargo.toml).
+For convenience there is also a feature named `performance` which enables all other features that improve overall performance.
+
+Most `cargo` commands accept the `--features` flag followed by a list of features to enable, e.g.
+```sh
+cargo build --features mimalloc,stack-array
+```
+
+## Testing
+
+- Rust tests
+    ```sh
+    cargo test
+    ```
+- Go tests
+    ```sh
+    cargo build --release # OR run `make` or `make tosca-rust` in Tosca directory
+    go test ../go/...
+    ```
+- CT
+    ```sh
+    cargo build --release # TODO remove when https://github.com/Fantom-foundation/Tosca/pull/778 is merged
+    cd ..
+    make
+    go run ./go/ct/driver run --full-mode evmrs
+    ```
+
+Also see [test.sh](./scripts/test.sh) which runs the Go tests and CT in full mode.
+
+## Benchmarking
```

Comment:
Can you describe how to compare benchmarks and a brief explanation of the output of this command? 
use Tosca/BUILD.md for inspiration.

Reply:
I'll just link to Tosca/BUILD.md#running-benchmarks it that is fine with you.

Reply:
can you compare the outputs using benchstat?

Reply:
Yes with a workaround. Have documented it now.